tools/cachemgr.cc

```diff
@@ -1086,20 +1086,29 @@ static void
 make_pub_auth(cachemgr_request * req)
 {
     static char buf[1024];
+    char *rfc1738_username = NULL, *rfc1738_passwd = NULL;
+
     safe_free(req->pub_auth);
     debug("cmgr: encoding for pub...\n");
 
     if (!req->passwd || !strlen(req->passwd))
         return;
 
+    /*rfc1738_escape is not reentrant */
+    rfc1738_username = xstrdup(rfc1738_escape(safe_str(req->user_name)));
+    rfc1738_passwd = xstrdup(rfc1738_escape(req->passwd));
```

Comment:
Please update this to C++11 style. This is not C code.

```suggestion
    auto *rfc1738_username = xstrdup(rfc1738_escape(safe_str(req->user_name)));
    auto *rfc1738_passwd = xstrdup(rfc1738_escape(req->passwd));
```

Reply:
Just to double-check, this is the CGI code "cachemgr.cc" which seems to be C and "auto *" isn't used anywhere else in that code. Can you please confirm?

Reply:
All `.cc` files are treated as C++ code. In most cases, we use modern C++ features when we add or change code, and cachemgr.cc is one of those neglected code areas that is rarely enhanced or touched... You do not _have to_ use `auto` in this PR -- such polishing is optional -- but please do it if you can.